app/utils/useDebounced.ts

```diff
@@ -0,0 +1,13 @@
+import { debounce } from 'lodash';
+import { useMemo } from 'react';
+
+const useDebouncedCallback = <T extends (...args: any) => any>(
```

Comment:
Agree, but `unknown` gives compiler-errors :grimacing: 

The implementation is (cooked) from lodash's types: Ended up using this since we truly allow any function, and once a function is passed will typescript infer and apply rules according to those arguments/return type.